src/main/scala/dx/core/ir/Type.scala

```diff
@@ -25,6 +25,8 @@ object Type {
     */
   case class TOptional(t: Type) extends Type
 
+  sealed trait TCollection extends Type
```

Comment:
Correct me if I'm wrong but TCollection doesn't seem to used anywhere so why is it introduced? Isn't it simpler for these types to be left Type?

Reply:
I can remove it here. It's another thing that's needed for Archive format, but I will add it in that branch.